model/secret.go

```diff
@@ -19,6 +19,12 @@ package model
 // the data immediately to the client of this interface.
 // TODO: add controller hooks for notification about changes to the secret
 type SecretRegistry interface {
-	// GetSecret retrieves the secret data by a platform specific URI
-	GetSecret(uri string) (map[string][]byte, error)
+	// GetTLSSecret retrieves a TLS secret by namespace and host
+	GetTLSSecret(namespace, host string) (*TLSSecret, error)
```

Comment:
I agree with @kyessenov . Namespace is a very k8s specific thing. A generic URI provides more flexibility in how platform specific nuances can be encoded in the URI.